### PR TITLE
feat(events):updation and deletion of events

### PR DESCRIPTION
**Pull request by BlesyMathew**
Signed-off-by: BLESSY MATHEW <74411288+BlesyMathew@users.noreply.github.com>

### CODE REVIEW

**Comment by shyluthomas** [comment]
@BlesyMathew 
Consistent commit messages help maintain a clean and understandable codebase, especially for future reference.
Here's a quick rundown of the guidelines:
Subject Line:
Start with a capitalized verb (e.g., "Fix", "Add", "Remove").
Briefly describe the change made in the commit.
Body:
Provide additional details about the change, if necessary.
Example:
Subject: Fix login form validation error (closes #123)

Body:
- Updated validation logic to handle empty username fields.
- Improved error message clarity.
Consider linking to the official GitHub commit message guidelines (https://git-scm.com/docs/git-commit) for more details.
Let's all strive for clear and informative commit messages!

File: src/pages/events/events.tsx [hunk 1]
```diff
@@ -7,20 +7,47 @@ import {
...
+} from "@/store/reducers/userReducer";
 import { useAppDispatch, useAppSelector } from "@/hooks/hooks";
 import { useEffect, useState } from "react";
 
 import { Button } from "@/components/ui/button";
 import OpenEventForm from "./openEventForm";
 import { toast } from "@/components/ui/use-toast";
 
+export const initialData = {
+  // id: "",
```
Comment (shyluthomas) [inline src/pages/events/events.tsx]: remove comments

File: src/pages/events/openEventForm.tsx [hunk 2]
```diff
@@ -6,36 +6,88 @@ import {
   AlertDialogHeader,
   AlertDialogTitle,
 } from "../../components/ui/alert-dialog";
+// import { initialData } from "./events";
```
Comment (shyluthomas) [inline src/pages/events/openEventForm.tsx]: remove commented

File: src/pages/events/openEventForm.tsx [hunk 3]
```diff
@@ -6,36 +6,88 @@ import {
...
+      dispatch(updateEvent({ ...data, id }));
+    }
+
     setEventForm(false);
     reset();
   };
   const data2 = useAppSelector((state) => state.userReducer.getProfile);
-
   const ownerId = data2.profile.data.user.id;
+  // dispatch(getPatchData(ownerId));
```
Comment (shyluthomas) [inline src/pages/events/openEventForm.tsx]: //

File: src/pages/events/openEventForm.tsx [hunk 4]
```diff
@@ -6,36 +6,88 @@ import {
...
+  const getUser = useAppSelector((state) => state.userReducer.patchResponse);
+  console.log(getUser);
```
Comment (shyluthomas) [inline src/pages/events/openEventForm.tsx]: clg

File: src/pages/events/openEventForm.tsx [hunk 5]
```diff
@@ -102,20 +153,35 @@ function OpenEventForm({ eventForm, setEventForm }: OpenEventFormProps) {
           <Input
             id="schedule"
             placeholder="Schedule"
-            {...register("eventItenary[0].schedule", { required: true })}
+            {...register("eventItenary.0.schedule", { required: true })}
```
Comment (shyluthomas) [inline src/pages/events/openEventForm.tsx:143]: truthy.  need to check (?) or &

File: src/pages/events/openEventForm.tsx [hunk 6]
```diff
@@ -102,20 +153,35 @@ function OpenEventForm({ eventForm, setEventForm }: OpenEventFormProps) {
...
+                <Input
+                  {...field}
+                  id="file"
+                  type="file"
+                  placeholder="File"
+                  value={value?.fileName}
+                  onChange={(event) => {
+                    onChange(event.target.files?.[0]);
+                  }}
+                  // {...register("file", { required: true })}
```
Comment (shyluthomas) [inline src/pages/events/openEventForm.tsx]: //

File: src/store/reducers/userReducer.ts [hunk 7]
```diff
@@ -85,6 +87,7 @@ const initialState = {
   idData: {
     id: "",
   },
+  delete: { id: "" },
```
Comment (shyluthomas) [inline src/store/reducers/userReducer.ts]: id is not string

File: src/store/saga/mainSaga.ts [hunk 8]
```diff
@@ -82,7 +82,7 @@ function* _getProfile(): any {
   }
 }
 
-function* _getPatchData(action: any): any {
+function* _getPatchData(action: ProfileId): any {
```
Comment (shyluthomas) [inline src/store/saga/mainSaga.ts]: profileId: use generic type

**Review by shyluthomas — CHANGES_REQUESTED** [review]

File: src/store/saga/mainSaga.ts (hunk 8 above)
Comment (BlesyMathew) [reply src/store/saga/mainSaga.ts]: can you provide an example ?

File: src/store/reducers/userReducer.ts (hunk 7 above)
Comment (BlesyMathew) [reply src/store/reducers/userReducer.ts]: I have changed the initial state to undefined

**Review by shyluthomas — APPROVED** [review]

**Pull request merged**